src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -229,6 +222,97 @@ MultiLogicalPlanCreate(Query *originalQuery, Query *queryTree,
 }
 
 
+/*
+ * ShouldUseSubqueryPushDown determines whether it's desirable to use
+ * subquery pushdown to plan the query based on the original and
+ * rewritten query.
+ */
+static bool
```

Comment:
I like to have a function for this purpose. We can move things to pushdown planner one by one. For example first, we add `HAS_OUTER_JOIN` here etc and gradually utilize pushdown abilities.

Reply:
Yep, that's a great idea, though we should probably figure out what causes the failures in https://github.com/citusdata/citus/pull/1824